src/vs/workbench/contrib/chat/browser/chat.ts

```diff
@@ -134,3 +137,13 @@ export interface IChatWidget {
 export interface IChatViewPane {
 	clear(): void;
 }
+
+
+export interface ICodeBlockActionContextProvider {
+	getCodeBlockContext(editor?: ICodeEditor): ICodeBlockActionContext | undefined;
+}
+export interface ICodeBlockContextProviderService {
```

Comment:
ChatCodeBlockContextProviderService... so it's more obvious in the global scope?